File: skills/gaode-heatmap/scripts/poi_search.py
#!/usr/bin/env python3
"""
高德 POI 搜索脚本
用于搜索门店、竞品、餐厅等兴趣点数据
"""
import requests
import json
import sys

# 从 apis.md 读取 API Key
API_KEY = "5edd3a2f42e77bec526e0b0bf9e0a9bf"
BASE_URL = "https://restapi.amap.com/v3/place/text"

def search_poi(keywords, city, page=1, offset=25):
    """
    搜索 POI 数据
    
    Args:
        keywords: 搜索关键词（如"茅台专卖店"、"餐厅"）
        city: 目标城市（如"贵阳市"）
        page: 页码（默认 1）
        offset: 每页数量（最大 25）
    
    Returns:
        dict: POI 搜索结果
    """
    params = {
        "key": API_KEY,
        "keywords": keywords,
        "city": city,
        "output": "json",
        "offset": offset,
        "page": page
    }
    
    response = requests.get(BASE_URL, params=params)
    return response.json()

def extract_locations(poi_result):
    """
    从 POI 结果中提取经纬度数据
    
    Args:
        poi_result: POI 搜索结果的 JSON
    
    Returns:
        list: [[经度，纬度，权重], ...]
    """
    locations = []
    if poi_result.get("pois"):
        for poi in poi_result["pois"]:
            if "location" in poi:
                lng, lat = poi["location"].split(",")
                locations.append([float(lng), float(lat), 1])
    return locations

def main():
    if len(sys.argv) < 3:
        print("用法：python poi_search.py <关键词> <城市>")
        print("示例：python poi_search.py 茅台专卖店 贵阳市")
        sys.exit(1)
    
    keywords = sys.argv[1]
    city = sys.argv[2]
    
    print(f"搜索：{keywords} - {city}")
    result = search_poi(keywords, city)
    
    if result.get("status") == "1":
        count = result.get("count", 0)
        print(f"找到 {count} 条记录")
        
        # 输出前 10 条详情
        if result.get("pois"):
            print(f"\n前 10 条结果：")
            for i, poi in enumerate(result["pois"][:10], 1):
                print(f"{i}. {poi.get('name', 'N/A')} - {poi.get('address', 'N/A')}")
    else:
        print(f"搜索失败：{result.get('info', 'Unknown error')}")

if __name__ == "__main__":
    main()

File: skills/amap-lbs-skill/scripts/poi-search.js
#!/usr/bin/env node

/**
 * POI 搜索脚本
 * 使用方法:
 * node scripts/poi-search.js --keywords=肯德基 --city=北京
 * 或设置环境变量: AMAP_KEY=your_key node scripts/poi-search.js --keywords=肯德基
 */

const { searchPOI } = require('../index');

// 解析命令行参数
function parseArgs() {
  const args = {};
  process.argv.slice(2).forEach(arg => {
    if (arg.startsWith('--')) {
      const [key, value] = arg.slice(2).split('=');
      args[key] = value;
    }
  });
  return args;
}

// 主函数
async function main() {
  const args = parseArgs();
  
  // 检查必需参数
  if (!args.keywords) {
    console.error('❌ 缺少必需参数: --keywords');
    console.log('\n使用方法:');
    console.log('node scripts/poi-search.js --keywords=关键词 [--city=城市] [--types=类型] [--page=页码] [--offset=每页数量]');
    console.log('\n示例:');
    console.log('node scripts/poi-search.js --keywords=肯德基 --city=北京 --page=1 --offset=20');
    process.exit(1);
  }
  
  // 检查是否设置了 AMAP_KEY
  if (!process.env.AMAP_KEY) {
    console.error('❌ 请设置环境变量 AMAP_KEY');
    console.log('\n示例:');
    console.log('export AMAP_KEY=your_amap_key');
    console.log('node scripts/poi-search.js --keywords=美食 --location=116.397428,39.90923 --radius=1000');
    process.exit(1);
  }
  
  // 构建搜索参数
  const params = {
    keywords: args.keywords,
    city: args.city || '',
    types: args.types || '',
    page: parseInt(args.page) || 1,
    offset: parseInt(args.offset) || 10
  };
  
  // 可选参数
  if (args.location) params.location = args.location;
  if (args.radius) params.radius = parseInt(args.radius);
  if (args.cityLimit !== undefined) params.cityLimit = args.cityLimit === 'true';
  
  try {
    const result = await searchPOI(params);
    
    if (result && result.pois && result.pois.length > 0) {
      console.log(`\n📍 共找到 ${result.count} 条结果，当前显示第 ${params.page} 页:\n`);
      console.log('='.repeat(80));
      
      result.pois.forEach((poi, index) => {
        const num = (params.page - 1) * params.offset + index + 1;
        console.log(`\n${num}. ${poi.name}`);
        console.log(`   📍 地址: ${poi.address || '无'}`);
        console.log(`   🏷️  类型: ${poi.type}`);
        console.log(`   📞 电话: ${poi.tel || '无'}`);
        console.log(`   🗺️  坐标: ${poi.location}`);
        if (poi.distance) {
          console.log(`   📏 距离: ${poi.distance}米`);
        }
      });
      
      console.log('\n' + '='.repeat(80));
      
      // 输出分页信息
      const totalPages = Math.ceil(result.count / params.offset);
      console.log(`\n第 ${params.page}/${totalPages} 页`);
      
      if (params.page < totalPages) {
        console.log(`\n💡 查看下一页: node scripts/poi-search.js --keywords=${params.keywords} --city=${params.city} --page=${params.page + 1}`);
      }
    } else {
      console.log('\n❌ 未找到相关结果');
    }
  } catch (error) {
    console.error('\n❌ 执行失败:', error.message);
    process.exit(1);
  }
}

// 执行主函数
main();
